records_mover/records/targets/data_url.py

```diff
@@ -63,25 +60,6 @@ def move_from_records_directory(self,
                               records_format.generate_filename('data'): self.output_loc.url
                           })
 
-    def move_from_temp_loc_after_filling_it(self,
-                                            records_source:
-                                            SupportsMoveToRecordsDirectory,
-                                            processing_instructions:
-                                            ProcessingInstructions) -> MoveResult:
-        pis = processing_instructions
-        records_format = records_source.compatible_format(self)
-        if records_format is None:
-            raise NotImplementedError("No compatible records format between "
-                                      f"{records_source} and {self}")
-        with self.output_loc.temporary_directory() as temp_loc:
-            directory = RecordsDirectory(records_loc=temp_loc)
-            records_source.\
-                move_to_records_directory(directory,
-                                          records_format=records_format,
-                                          processing_instructions=pis)
-            return self.move_from_records_directory(directory,
-                                                    processing_instructions)
-
```

Comment:
This function said "Move something to this filesystem by first creating a temporary directory ON THIS SAME FILESYSTEM and then copying from there".  Kind of silly, since you can also just copy directly to the intended destination (see the 

The purpose of `move_from_temp_loc_after_filling_it()` is for systems like Redshift where the fastest way to copy things in is from an S3 bucket.  In those cases, even if the source isn't an S3 bucket, we want to copy to a temporary S3 bucket before loading.

I'm guessing this function in this file was implemented at some point of time in the past in order to work around some wart or another in the `move()` algorithm.  The `move()` algorithm uses these Python interfaces to figure out a way to get data from the source to the target in a multiple dispatch way.  

There's been a lot of incremental development and refactoring among all of these things, and evolutionary dead-ends are not unknown :)